Funcky.Test/Extensions/StringExtensionsTest.cs

```diff
@@ -0,0 +1,85 @@
+            AssertIsSome(NeedlePosition, Haystack.IndexOfOrNone(ExistingNeedleChar, StringComparison.InvariantCulture));
+            AssertIsSome(NeedlePosition, Haystack.IndexOfOrNone(ExistingNeedleChar, startIndex: 0, count: Haystack.Length));
+            AssertIsSome(NeedlePosition, Haystack.IndexOfOrNone(ExistingNeedle));
+            AssertIsSome(NeedlePosition, Haystack.IndexOfOrNone(ExistingNeedle, startIndex: 0));
+            AssertIsSome(NeedlePosition, Haystack.IndexOfOrNone(ExistingNeedle, startIndex: 0, count: Haystack.Length));
+            AssertIsSome(NeedlePosition, Haystack.IndexOfOrNone(ExistingNeedle, StringComparison.InvariantCulture));
+            AssertIsSome(NeedlePosition, Haystack.IndexOfOrNone(ExistingNeedle, startIndex: 0, StringComparison.InvariantCulture));
+            AssertIsSome(NeedlePosition, Haystack.IndexOfOrNone(ExistingNeedle, startIndex: 0, count: Haystack.Length, StringComparison.InvariantCulture));
+            AssertIsSome(NeedlePosition, Haystack.IndexOfAnyOrNone(new[] { ExistingNeedleChar }));
+            AssertIsSome(NeedlePosition, Haystack.IndexOfAnyOrNone(new[] { ExistingNeedleChar }, startIndex: 0));
+            AssertIsSome(NeedlePosition, Haystack.IndexOfAnyOrNone(new[] { ExistingNeedleChar }, startIndex: 0, count: Haystack.Length));
+            AssertIsSome(NeedlePosition, Haystack.LastIndexOfOrNone(ExistingNeedleChar));
+            AssertIsSome(NeedlePosition, Haystack.LastIndexOfOrNone(ExistingNeedleChar, startIndex: Haystack.Length - 1));
+            AssertIsSome(NeedlePosition, Haystack.LastIndexOfOrNone(ExistingNeedleChar, startIndex: Haystack.Length - 1, count: Haystack.Length));
+            AssertIsSome(NeedlePosition, Haystack.LastIndexOfOrNone(ExistingNeedle));
+            AssertIsSome(NeedlePosition, Haystack.LastIndexOfOrNone(ExistingNeedle, startIndex: Haystack.Length - 1));
+            AssertIsSome(NeedlePosition, Haystack.LastIndexOfOrNone(ExistingNeedle, startIndex: Haystack.Length - 1, count: Haystack.Length));
+            AssertIsSome(NeedlePosition, Haystack.LastIndexOfOrNone(ExistingNeedle, StringComparison.InvariantCulture));
+            AssertIsSome(NeedlePosition, Haystack.LastIndexOfOrNone(ExistingNeedle, startIndex: Haystack.Length - 1, StringComparison.InvariantCulture));
+            AssertIsSome(NeedlePosition, Haystack.LastIndexOfOrNone(ExistingNeedle, startIndex: Haystack.Length - 1, count: Haystack.Length, StringComparison.InvariantCulture));
+            AssertIsSome(NeedlePosition, Haystack.LastIndexOfAnyOrNone(new[] { ExistingNeedleChar }));
+            AssertIsSome(NeedlePosition, Haystack.LastIndexOfAnyOrNone(new[] { ExistingNeedleChar }, startIndex: Haystack.Length - 1));
+            AssertIsSome(NeedlePosition, Haystack.LastIndexOfAnyOrNone(new[] { ExistingNeedleChar }, startIndex: Haystack.Length - 1, count: Haystack.Length));
+        }
+
+        private static void AssertIsNone<TItem>(Option<TItem> option)
```

Comment:
Maybe extract the Test utilities into it's own class, or to the FunctionalAsserts?

Reply:
636e77a29721a27d88d57e2e0f38a9a9317ab2bf and a652dbdca179c5ad32868918ee152f7790846579